src/Symfony/Component/Form/Extension/Core/DataTransformer/IntlTimeZoneToStringTransformer.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Extension\Core\DataTransformer;
+
+use Symfony\Component\Form\DataTransformerInterface;
+use Symfony\Component\Form\Exception\TransformationFailedException;
+
+/**
+ * Transforms between a timezone identifier string and a IntlTimeZone object.
+ *
+ * @author Roland Franssen <franssen.roland@gmai.com>
+ */
+class IntlTimeZoneToStringTransformer implements DataTransformerInterface
+{
+    private $multiple;
+
+    public function __construct(bool $multiple = false)
+    {
+        $this->multiple = $multiple;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function transform($intlTimeZone)
+    {
+        if (null === $intlTimeZone) {
+            return;
+        }
+
+        if ($this->multiple) {
+            if (!\is_array($intlTimeZone)) {
+                throw new TransformationFailedException('Expected an array.');
```

Comment:
Thinking about a developer who sees this error message, I wonder if we should add more info to it to make it easier to understand. Example:

```
`Expected an array of \IntlTimeZone objects.`
```